sopel/coretasks.py

```diff
@@ -106,6 +112,12 @@ def _handle_sasl_capability(
             'cannot authenticate with SASL.',
         )
         return plugin.CapabilityNegotiation.ERROR
+    elif auth_target not in ["PLAIN", "EXTERNAL", "SCRAM-SHA-256"]:
+        LOGGER.error(
+            'Configured SASL method %r is not supported by Sopel.',
+            auth_target,
+        )
+        return plugin.CapabilityNegotiation.ERROR
```

Comment:
In my opinion there are better places to check this:

* few lines down in the file we check that the configured SASL method is supported by the IRC server
* the configuration option can have validation
* maybe the setup function of coretasks could check for that

Reply:
Do we care about failing the config if the server doesn't support SASL at all? If yes, we should also error on SASL configured but no server support. If no, I've made the check consistent with server mech mismatch.